### PR TITLE
[Sqlparser] Support KILL statements

### PR DESCRIPTION
**Pull request by danieltahara**
Add support for KILL and KILL QUERY to allow clients to perform
"context" cancellation.

Signed-off-by: Daniel Tahara <tahara@dropbox.com>

### CODE REVIEW

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -163,6 +163,7 @@ func forceEOF(yylex interface{}) {
 %token <bytes> MAXVALUE PARTITION REORGANIZE LESS THAN PROCEDURE TRIGGER
 %token <bytes> VINDEX VINDEXES
 %token <bytes> STATUS VARIABLES
+%token <bytes> KILL
```
Comment (sougou) [inline go/vt/sqlparser/sql.y]: Need to add this to the list of non-reserved keywords.

**Comment by xhh1989** [comment]
a kill statement may run on different vtgate?

**Comment by sougou** [comment]
I believe this is a parser-only addition for non-vitess users.

**Comment by sougou** [comment]
@danieltahara do you still want this? If so, the review comment needs to be addressed, and conflicts resolved.

**Comment by danieltahara** [comment]
We don't actually need it with any level of urgency, so I just haven't gotten around to working on it. Feel free to close if you want.

**Pull request closed without merging**